chef-config/lib/chef-config/package_task.rb

```diff
@@ -0,0 +1,223 @@
+#
+# Author:: Kartik Null Cating-Subramanian (<ksubramanian@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'rake'
+require 'rubygems'
+require 'rubygems/package_task'
+
+module ChefConfig
+  class PackageTask < Rake::TaskLib
+
+    # Full path to root of top-level repository.  All other files (like VERSION or
+    # lib/<module_path_name>/version.rb are rooted at this path).
+    attr_accessor :root_path
+
+    # Name of the top-level module/library build built.  This is used to define
+    # the top level module which contains VERSION and MODULE_ROOT.
+    attr_accessor :module_name
+
+    # Should the generated version.rb be in a class or module?  Default is false (module).
+    attr_accessor :generate_version_class
+
+    # Paths to the roots of any components that also support ChefPackageTask.
+    # If relative paths are provided, they are rooted against root_path.
+    attr_accessor :component_paths
+
+    # This is the module name as it appears on the path "lib/module/".
+    # e.g. for module_name  "ChefDK", you'd want module_path_name to be "chef-dk".
+    # The default is module_name but lower-cased.
+    attr_writer :module_path_name
```

Comment:
This name is used in gem-uninstall - so it might not work there.  I'll rename it but as we run into it, we might need to simply split module_path and take the last fragment for the uninstall task.